nats-base-client/protocol.ts

```diff
@@ -603,10 +604,17 @@ export class ProtocolHandler implements Dispatcher<ParserEvent> {
     this.outbound.fill(buf, ...payloads);
 
     if (len === 0) {
-      setTimeout(() => {
+      this.flusher = setTimeout(() => {
         this.flushPending();
       });
     } else if (this.outbound.size() >= this.pendingLimit) {
+      // if we have a flusher, clear it - otherwise in a bench
+      // type scenario where the main loop is dominated by a publisher
+      // we create many timers.
+      if (this.flusher) {
+        clearTimeout(this.flusher);
+        this.flusher = undefined;
```

Comment:
Nit: nullable type, so `null`? (strict null check would disallow this).

```suggestion
        this.flusher = null;
```

Reply:
fixed